onnx/defs/tensor/defs.cc

```diff
@@ -3890,4 +3890,78 @@ ONNX_OPERATOR_SET_SCHEMA(
 	        }
         }));
 
+static const char* Col2Im_ver16_doc = R"DOC(
+The operator rearranges the input matrix columns into blocks.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+	Col2Im,
+	16,
+	OpSchema()
+	.Attr(
+		"dilation",
+		"Dilation value along each spatial axis of the kernel."
+    "If not present, the dilation defaults to 1 for each spatial axis",
+    AttributeProto::INTS,
+    OPTIONAL_VALUE)
+	.Attr(
+		"padding",
+		"Padding for the beginning and ending along each spatial axis"
+    "If not present, the padding defaults to 0 along start and end of each spatial axis.",
+    AttributeProto::INTS,
+    OPTIONAL_VALUE)
+	.Attr(
+		"stride",
+		"Stride along each spatial axis."
+    "If not present, the stride faults to 1 for each spatial axis.",
+    AttributeProto::INTS,
+    OPTIONAL_VALUE)
+	.SetDoc(Col2Im_ver16_doc)
+	.Input(
+		0,
+		"input",
+		"Input data tensor to be rearranged into blocks.",
+		"T",
+		OpSchema::Single,
+		true,
+		1,
+		OpSchema::Differentiable)
+	.Input(
+		1,
+		"output_size",
+		"The shape of the input before applying im2col transformation.",
+		"tensor(int64)",
+		OpSchema::Single,
+		true,
+		1,
+		OpSchema::NonDifferentiable)
+	.Input(
+		2,
+		"kernel_size",
+		"The shape of the kernel to apply on the input.",
+		"tensor(int64)",
+		OpSchema::Single,
+		true,
+		1,
+		OpSchema::NonDifferentiable)
+	.Output(
+		0,
+		"output",
+		"Output tensor produced by rearranging matrix into blocks.",
+		"T",
+		OpSchema::Single,
+		true,
+		1,
+		OpSchema::Differentiable)
+  .TypeConstraint(
+            "T",
+            OpSchema::all_tensor_types_with_bfloat(),
+            "Constrain input and output types to all tensor types.")
+	.TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+    propagateElemTypeFromInputToOutput(ctx, 0, 0);
+    if (hasNInputShapes(ctx, 1)) {
+      propagateShapeFromInputToOutput(ctx, 0, 0);
```

Comment:
this is not right... you are setting the shape of output to be same as shape of input.